src/wp-includes/class-wp-scripts.php

```diff
@@ -281,6 +281,24 @@ public function do_item( $handle, $group = false ) {
 			$intended_strategy = '';
 		}
 
+		// If the script is in the header group, current output is the header, the intended strategy is `defer`, the
+		// actual strategy is not defer and all dependent scripts are in the footer - move this script to the footer.
+		if ( 'defer' === $intended_strategy && 'defer' !== $strategy && 0 === $this->groups[ $handle ] && 0 === $group ) {
```

Comment:
Is there a reason why this is limited to `defer`? Shouldn't it also apply to `async`?

```suggestion
		if ( $intended_strategy && ! $this->is_delayed_strategy( $strategy ) && 0 === $this->groups[ $handle ] && 0 === $group ) {
```

Reply:
good point